src/Pim/Bundle/ApiBundle/Controller/ProductController.php

```diff
@@ -302,7 +302,7 @@ public function partialUpdateAction(Request $request, $code)
 
         if ($isCreation) {
             $this->validateCodeConsistency($code, $data);
-            $product = $this->productBuilder->createProduct();
+            $product = $this->productBuilder->createProduct($code);
```

Comment:
as product identifier can be in the query path and/or in the body of the request, you should put ` $data['identifier'] = array_key_exists('identifier', $data) ? $data['identifier'] : $code;` before the condition `if ($isCreation)` and set `$product = $this->productBuilder->createProduct($data['identifier']);`